app/src/main/java/at/aau/serg/logic/StoreToken.kt

```diff
@@ -2,18 +2,52 @@ package at.aau.serg.logic
 
 import android.content.ContextWrapper
 import org.json.JSONException
+import org.json.JSONObject
 
-class StoreToken {
-    fun storeTokens(accessToken: String, refreshToken: String, context: ContextWrapper, secret: Secret){
-        try{
-            val sharedPreferences =secret.getSecretSharedPref(context)
-            val editor = sharedPreferences.edit()
-            editor.putString("accessToken", accessToken)
-            editor.putString("refreshToken", refreshToken)
-            editor.apply()
-        }catch (e: JSONException) {
-            e.printStackTrace()
-        }
+class StoreToken(private val context: ContextWrapper) {
+    private var secret: Secret = Secret()
```

Comment:
You should be able to mock the `Secret()` call instead of using the setter. For reference, take a look at the IO mock in `SocketHandlerTest.kt`.

Reply:
ty, I made it